src/test/java/org/isf/disease/test/Tests.java

```diff
@@ -183,6 +190,194 @@ public void testIoIsDescriptionPresent() throws Exception {
 		assertThat(result).isTrue();
 	}
 
+	@Test
+	public void testMgrGetDiseaseByCode() throws Exception {
+		String code = _setupTestDisease(false);
+		Disease foundDisease = diseaseBrowserManager.getDiseaseByCode(Integer.valueOf(code));
+		testDisease.check(foundDisease);
+	}
+
+	@Test
+	public void testMgrGetDiseases() throws Exception {
+		String code = _setupTestDisease(false);
+		Disease foundDisease = diseaseBrowserManager.getDiseaseByCode(Integer.valueOf(code));
+
+		ArrayList<Disease> diseases = diseaseBrowserManager.getDisease(foundDisease.getType().getCode());
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDiseaseOpd(foundDisease.getType().getCode());
+		assertThat(diseases).doesNotContain(foundDisease);
+		foundDisease.setOpdInclude(true);
+		diseaseIoOperationRepository.saveAndFlush(foundDisease);
+		diseases = diseaseBrowserManager.getDiseaseOpd(foundDisease.getType().getCode());
+		assertThat(diseases).contains(foundDisease);
+
+		foundDisease = diseaseBrowserManager.getDiseaseByCode(Integer.valueOf(code));
+		diseases = diseaseBrowserManager.getDiseaseIpdIn();
+		assertThat(diseases).doesNotContain(foundDisease);
+		foundDisease.setOpdInclude(true);
+		foundDisease.setIpdInInclude(true);
+		diseaseIoOperationRepository.saveAndFlush(foundDisease);
+		diseases = diseaseBrowserManager.getDiseaseOpd();
+		assertThat(diseases).contains(foundDisease);
+
+		foundDisease = diseaseBrowserManager.getDiseaseByCode(Integer.valueOf(code));
+		diseases = diseaseBrowserManager.getDiseaseIpdOut(foundDisease.getType().getCode());
+		assertThat(diseases).doesNotContain(foundDisease);
+		foundDisease.setOpdInclude(true);
+		foundDisease.setIpdInInclude(true);
+		foundDisease.setIpdOutInclude(true);
+		diseaseIoOperationRepository.saveAndFlush(foundDisease);
+		diseases = diseaseBrowserManager.getDiseaseIpdOut();
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDiseaseIpdIn();
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDiseaseIpdOut(foundDisease.getType().getCode());
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDiseaseIpdIn(foundDisease.getType().getCode());
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDiseaseAll();
+		assertThat(diseases).contains(foundDisease);
+
+		diseases = diseaseBrowserManager.getDisease();
+		assertThat(diseases).contains(foundDisease);
```

Comment:
Just asking: should not be tested separately?

Reply:
I could do that if you want; just thought that since the tests were so simple and they could all use the same setup I just lumped them together.   @mwithi If you want I'll split them out....

Reply:
The more I look at what I did I think it makes sense to keep them together as they are testing the various In/out/all conditions.

Reply:
Re: Junit 4 and parameterized tests can be done; see this [PR](https://github.com/informatici/openhospital-core/pull/239)

Reply:
@dbmalkovsky if we split the test we should also choose name method in a way similar to what @tomek39856 is suggesting? I think we can leave it as is now and refactor once we take a shared decision on Discussion Page → [Test naming convention](https://openhospital.atlassian.net/wiki/spaces/OHC/pages/1441071132/Test+naming+convention).

As for now we can just apply fix to typos and merge.